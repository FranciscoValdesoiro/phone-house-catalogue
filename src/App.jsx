import React from 'react';
import './App.css';
import DetailContainer from './components/DetailContainer'
import PhoneListContainer from './components/PhoneListContainer'
import GuardedRoute from './components/GuardedRoute'
import HeaderComponent from './components/HeaderComponent'

import { Route, BrowserRouter as Router, Switch } from 'react-router-dom';
import { useSelector} from 'react-redux'

const App = () => {

  const phoneSelected = useSelector(state => state.phones.PhoneSelected)
  const isPhoneSelected = Object.keys(phoneSelected).length > 0;

  return (
    <Router>
        <HeaderComponent img="gs-logo.png"/>
        <Switch>
            <Route path="/" exact>
                <PhoneListContainer />
            </Route>
            <GuardedRoute 
              auth={isPhoneSelected}
              component={DetailContainer} 
              path="/detail/:id"
            />
        </Switch>
    </Router>
  );
}

export default App;
