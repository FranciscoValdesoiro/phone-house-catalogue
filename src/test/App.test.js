import React from 'react';
import {Provider} from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import App from '../App.jsx';
import { mount } from 'enzyme'
import '@testing-library/jest-dom';
import generateStore from '../redux/store';


describe('App', () => {
  
  generateStore()
  const mockStore = configureMockStore([thunk]);
  

  it('should render a startup component and inizialice the store', () => {
    const store = mockStore({
      phones: {
        PhoneSelected: {},
        PhoneList:  [  
          {
            id: 0,
            name: "iPhone 7",
            manufacturer: "Apple",
            description: "iPhone 7 dramatically improves the most important aspects of the iPhone experience. It introduces advanced new camera systems. The best performance and battery life ever in an iPhone. Immersive stereo speakers. The brightest, most colorful iPhone display. Splash and water resistance*. And it looks every bit as powerful as it is. This is iPhone 7.",
            color: "black",
            price: 769,
            imageFileName: "IPhone_7.png",
            screen: "4,7 inch IPS",
            processor: "A10 Fusion",
            ram: 2
          }
        ],
      }
             
    })
    
    const wrapper = mount( <Provider store={store}><App/></Provider>)
    expect( wrapper ).toMatchSnapshot();
  })
});