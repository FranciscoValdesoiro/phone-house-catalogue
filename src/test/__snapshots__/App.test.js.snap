// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render a startup component and inizialice the store 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="header-inner"
    >
      <img
        alt="logo"
        src="/images/fv-logo.png"
      />
      <p />
    </div>
  </header>,
  <div>
    <div
      className="phones-container"
    >
      <a
        data-cy="phone_7-card"
        href="/detail/phone_7"
        onClick={[Function]}
      >
        <div
          className="card-container"
        >
          <h3>
            iPhone 7
          </h3>
          <div
            className="content"
          >
            <img
              alt="iPhone 7phone image"
              className="phone-image"
              src="/images/IPhone_7.png"
            />
          </div>
          <p>
            Click me for more details
          </p>
        </div>
      </a>
    </div>
  </div>,
]
`;
